File: Clean-Architecture/Feature/dart/example/lib/features/notifications/presentation/controllers/notification_controller.dart
import 'package:flutter/foundation.dart';

import '../../domain/entities/notification_request.dart';
import '../../domain/usecases/send_notification.dart';

class NotificationController extends ChangeNotifier {
  NotificationController(this._sendNotification);
  final SendNotification _sendNotification;

  bool isLoading = false;
  String? message;

  Future<void> send({
    required String channel,
    required String recipient,
    required String text,
  }) async {
    isLoading = true;
    message = null;
    notifyListeners();

    final result = await _sendNotification(NotificationRequest(
      channel: channel,
      recipient: recipient,
      message: text,
    ));

    result.fold(
      onFailure: (failure) => message = failure.message,
      onSuccess: (notification) =>
          message = '${notification.channel}: ${notification.status} (${notification.id})',
    );

    isLoading = false;
    notifyListeners();
  }
}

File: Clean-Architecture/Feature/Javascript/example/src/features/notifications/presentation/controllers/NotificationController.js
import { validateRequired } from '../../../../core/validation/validateRequired.js';

export class NotificationController {
  constructor({ sendNotification }) {
    this.sendNotification = sendNotification;
  }

  send = async (req, res) => {
    validateRequired(req.body, ['channel', 'recipient', 'message']);
    const notification = await this.sendNotification.execute(req.body);
    res.status(201).json({ data: notification });
  };
}
